src/ErrorBoundary.ts

```diff
@@ -46,8 +56,30 @@ export class ErrorBoundary extends Component<
     }
   }
 
+  componentDidMount() {
+    window.addEventListener("error", handleSuppressLogging);
```

Comment:
If we had the `suppressLogging` prop, then I think this logic should be conditional:
```suggestion
    if (this.props.suppressLogging) {
      window.addEventListener("error", handleSuppressLogging);
    }
```

That being said, I don't think I understand the motivation for the `suppressLogging` prop.

Reply:
I agree that the logic should be conditional, and I did apply some condition triggers but in another life cycle method (`shouldComponentUpdate`). The reason behind this design decision lies in the case when multiple `ErrorBoundary`'s are present, e.g., 
```
<ErrorBoundary suppressLogging>
  <OutterChild />
  <ErrorBoundary>
    <InnerChild />
  </ErrorBoundary>
</ErrorBoundary>
```
In this case, the event listener will be mounted after all. However, if we still want the errors from `OutterChild` and from `InnerChild` to be treated differently, we need to add some more logic beyond the method of `componentDidMount`. Therefore, I decided to just  mount the event listener anyway in `componentDidMount`. The explanation of the logic in `shouldComponentUpdate` is in the comment above. Please let me know if you think this is over-engineered.

Reply:
Truthfully, I'm not very confident that we can move forward with this proposed change. For one, we can't use `shouldComponentUpdate` for side effects (as mentioned [here](https://github.com/bvaughn/react-error-boundary/pull/191/files#r1725002695)). This also feels pretty fragile, and I'm concerned about how it can impact the "break on uncaught exceptions" behavior.

Reply:
Thank you for pointing me to the [article](https://legacy.reactjs.org/docs/hooks-effect.html#example-using-classes). I examined the exposition and the example closely. From my understanding, the major concern there was that "*It would be too early*" (to implement the side effects.) But if we intentionally want to carry out the side effects at an early stage, wouldn't it justify the use of `shouldComponentUpdate`?

Reply:
Maybe [this blog post](https://legacy.reactjs.org/docs/strict-mode.html#detecting-unexpected-side-effects) will do a better job of explaining why side effects don't belong in a render-phase method like `shouldComponentUpdate`.

In this case, if `componentDidUpdate` is _too late_ to do what you want, then I think that indicates you maybe want to do the thing at the point where the action is triggered (when the show-boundary method is called).

That being said, as I mentioned in the comment above, I'm not really confident about landing this change at this point.

Reply:
> Maybe [this blog post](https://legacy.reactjs.org/docs/strict-mode.html#detecting-unexpected-side-effects) will do a better job of explaining why side effects don't belong in a render-phase method like `shouldComponentUpdate`.
> 
> In this case, if `componentDidUpdate` is _too late_ to do what you want, then I think that indicates you maybe want to do the thing at the point where the action is triggered (when the show-boundary method is called).
> 
> That being said, as I mentioned in the comment above, I'm not really confident about landing this change at this point.

That I can do the thing at the point where the action is triggered (when the show-boundary method is called) makes a great point. I will proceed along those lines to come up with another solution. Thanks.